src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.cs

```diff
@@ -676,7 +676,12 @@ public ArrayTypeSymbol Retarget(ArrayTypeSymbol type)
                     return type;
                 }
 
-                return new ArrayTypeSymbol(this.RetargetingAssembly, newElement, newModifiers, type.Rank);
+                if (type.IsSZArray)
+                {
+                    return ArrayTypeSymbol.CreateSZArray(this.RetargetingAssembly, newElement, newModifiers);
+                }
+
+                return ArrayTypeSymbol.CreateMDArray(this.RetargetingAssembly, newElement, type.Rank, newModifiers);
```

Comment:
No, it will convert MDArray of rank 1 to SZArray, we should preserve the kind here and in other similar places..